lib/Onyx.d.ts

```diff
@@ -0,0 +1,278 @@
+ * @param {String|Function} [mapping.selector] THIS PARAM IS ONLY USED WITH withOnyx(). If included, this will be used to subscribe to a subset of an Onyx key's data.
+ *       If the selector is a string, the selector is passed to lodashGet on the sourceData. If the selector is a function, the sourceData and withOnyx state are
+ *       passed to the selector and should return the simplified data. Using this setting on `withOnyx` can have very positive performance benefits because the component
+ *       will only re-render when the subset of data changes. Otherwise, any change of data on any property would normally cause the component to re-render (and that can
+ *       be expensive from a performance standpoint).
+ * @returns an ID to use when calling disconnect
+ */
+declare function connect<TKey extends Key | CollectionKey>(mapping: ConnectOptions<TKey>): number;
+
+/**
+ * Remove the listener for a react component
+ * @example
+ * Onyx.disconnect(connectionID);
+ *
+ * @param connectionID unique id returned by call to Onyx.connect()
+ * @param [keyToRemoveFromEvictionBlocklist]
+ */
+declare function disconnect<TKey extends Key | CollectionKey>(connectionID: number, keyToRemoveFromEvictionBlocklist?: TKey): void;
+
+/**
+ * Write a value to our store with the given key
+ *
+ * @param key ONYXKEY to set
+ * @param value value to store
+ */
+declare function set<TKey extends Key | CollectionKey>(key: TKey, value: Value[TKey] | null): Promise<void>;
+
+/**
+ * Sets multiple keys and values
+ *
+ * @example Onyx.multiSet({'key1': 'a', 'key2': 'b'});
+ *
+ * @param data object keyed by ONYXKEYS and the values to set
+ */
+declare function multiSet(data: Partial<KeyValueMap>): Promise<void>;
+
+/**
+ * Merge a new value into an existing value at a key.
+ *
+ * The types of values that can be merged are `Object` and `Array`. To set another type of value use `Onyx.set()`. Merge
+ * behavior uses lodash/merge under the hood for `Object` and simple concatenation for `Array`. However, it's important
+ * to note that if you have an array value property on an `Object` that the default behavior of lodash/merge is not to
+ * concatenate. See here: https://github.com/lodash/lodash/issues/2872
+ *
+ * Calls to `Onyx.merge()` are batched so that any calls performed in a single tick will stack in a queue and get
+ * applied in the order they were called. Note: `Onyx.set()` calls do not work this way so use caution when mixing
+ * `Onyx.merge()` and `Onyx.set()`.
+ *
+ * @example
+ * Onyx.merge(ONYXKEYS.EMPLOYEE_LIST, ['Joe']); // -> ['Joe']
+ * Onyx.merge(ONYXKEYS.EMPLOYEE_LIST, ['Jack']); // -> ['Joe', 'Jack']
+ * Onyx.merge(ONYXKEYS.POLICY, {id: 1}); // -> {id: 1}
+ * Onyx.merge(ONYXKEYS.POLICY, {name: 'My Workspace'}); // -> {id: 1, name: 'My Workspace'}
+ *
+ * @param key ONYXKEYS key
+ * @param value Object or Array value to merge
+ */
+declare function merge<TKey extends Key | CollectionKey>(key: TKey, value: PartialDeep<Value[TKey]>): Promise<void>;
```

Comment:
in the case of merge, we cannot pass `null`?

Reply:
`@param {(Object|Array)} value Object or Array value to merge` I don't think we should allow null here. Looking at implementation also suggest null shouldn't be allowed.